agent-inject/agent/annotations_test.go

```diff
@@ -1243,3 +1243,46 @@ func TestDisableKeepAlives(t *testing.T) {
 		})
 	}
 }
+
+func TestParseTelemetryAnnotations(t *testing.T) {
+	tests := map[string]struct {
+		annotations    map[string]string
+		expectedValues map[string]interface{}
+	}{
+		"prometheus": {
+			annotations: map[string]string{
+				"vault.hashicorp.com/agent-telemetry-prometheus_retention_time": "5s",
```

Comment:
I am surprised that the `json.Unmarshal` above works on this: `5s` isn't a valid JSON string (thought `"5s"` is).

I feel like we would need to have an escape hatch to treat the parameter as a string in the default case.

But that would presume that this test is failing, which it isn't?

I'm confused :)

Reply:
I *think* it's working the way it should? Maybe I'm not following why we would need the escape hatch?

Reply:
So, I guess I don't understand how this is working, honestly:

* The annotation is `5s`. Not `"5s"`.
* This gets passed as the `value` into: https://github.com/hashicorp/vault-k8s/pull/413/files#diff-8e04ce2f98b3799b387789c744675d1f7d5ad9751312f2deee564b461b9384eaR854
* That should return an error:

https://go.dev/play/p/ihsDQQbuO_h

Reply:
Okay, I am mistaken -- there already is an escape hatch. I was misreading the code. The `if` statement above is only triggered if `err != nil`, i.e., there was a JSON parsing error. I had everything kind of backwards.

Reply:
Annotations in kubernetes are always `map[string]string{}` so it will always get passed in as a string right? However, `[]byte{'5s'}` is not a string?

Reply:
Ahhhh, okay now I see why there was confusion! Sorry about that :D

Reply:
No worries, it was purely confusion on my end. :)